src/stenographer.c

```diff
@@ -0,0 +1,372 @@
+/**
+ * collectd - src/stenographer.c
+ * Copyright (C) 2015       Google
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
```

Comment:
The collectd code base is moving towards the ISC license (https://en.wikipedia.org/wiki/ISC_license). Would that be an option for you as well?